app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt

```diff
@@ -72,7 +76,12 @@ class ItemsViewModel @Inject constructor(
      */
     fun setItem(item: Item, callback: ((String?) -> Unit)? = null) {
         viewModelScope.launch(Dispatchers.IO) {
-            val itemId = itemRepository.set(item)
+            val uploadUrl = item.image?.let {
+                if(!it.startsWith("https://")){
+                    imageStore.store(it.toUri())
+                }
+            }
+            val itemId = itemRepository.set(item.copy(image = uploadUrl.toString()))
```

Comment:
You have to add an else where you return the current URL so you don't lose the existing image
```suggestion
            val uploadUrl = item.image?.let {
                if (!it.startsWith("https://")) {
                    imageStore.store(it.toUri()).toString()
                } else {
                    it
                }
            }
            val itemId = itemRepository.set(item.copy(image = uploadUrl))
```

Reply:
Yup thank you for the catch, modified and implemented in the latest commit